src/arch/cuda/hybridAPI/buddy_allocator.h

```diff
@@ -0,0 +1,247 @@
+#ifndef __BUDDY_ALLOCATOR_H_
```

Comment:
There seems to be quite a bit of code in this header file. It'll be good to just limit small functions and data structure definitions to headers and move other code to a `.C`.

Reply:
I was originally thinking of a header-only library but now I agree that it's better to move the implementation to a `.C`, will update